analyzers/src/SonarAnalyzer.Common/Rules/Utilities/TokenTypeAnalyzerBase.cs

```diff
@@ -172,27 +175,34 @@ symbol switch
 
         protected internal abstract class TriviaClassifierBase
         {
+            private readonly string filePath;
+
             protected abstract bool IsDocComment(SyntaxTrivia trivia);
             protected abstract bool IsRegularComment(SyntaxTrivia trivia);
 
+            protected TriviaClassifierBase(string filePath)
+            {
+                this.filePath = filePath;
+            }
+
             public TokenInfo ClassifyTrivia(SyntaxTrivia trivia) =>
-                trivia switch
-                {
-                    _ when IsRegularComment(trivia) => TokenInfo(trivia.SyntaxTree, TokenType.Comment, trivia.Span),
-                    _ when IsDocComment(trivia) => ClassifyDocComment(trivia),
-                    // Handle preprocessor directives here
-                    _ => null,
-                };
+                string.IsNullOrWhiteSpace(filePath)
```

Comment:
Also here, I think we should ignore tokens if they don't have a file path. Do you have maybe an example of when this happens?